object/src/mem.rs

```diff
@@ -570,6 +583,34 @@ mod tests {
         assert_eq!(elem1, &another_patient_name);
     }
 
+    #[test]
+    fn inmem_object_take_element() {
+        let another_patient_name = DataElement::new(
+            Tag(0x0010, 0x0010),
+            VR::PN,
+            PrimitiveValue::Str("Doe^John".to_string()).into(),
+        );
+        let mut obj = InMemDicomObject::create_empty();
+        obj.put(another_patient_name.clone());
+        let elem1 = obj.take_element(Tag(0x0010, 0x0010)).unwrap();
+        assert_eq!(elem1, another_patient_name);
+        assert!(obj.take_element(Tag(0x0010, 0x0010)).is_err());
+    }
+
+    #[test]
+    fn inmem_object_take_element_by_name() {
+        let another_patient_name = DataElement::new(
+            Tag(0x0010, 0x0010),
+            VR::PN,
+            PrimitiveValue::Str("Doe^John".to_string()).into(),
+        );
+        let mut obj = InMemDicomObject::create_empty();
+        obj.put(another_patient_name.clone());
+        let elem1 = obj.take_element_by_name("PatientName").unwrap();
+        assert_eq!(elem1, another_patient_name);
+        assert!(obj.take_element_by_name("PatientName").is_err());
```

Comment:
Changing this assertion will allow us to verify the issue above.

```suggestion
        assert!(matches!(
            obj.take_element_by_name("PatientName"),
            Err(Error::NoSuchDataElementAlias {
                tag: Tag(0x0010, 0x0010),
                alias,
                ..
            }) if alias == "PatientName")
        );
```

Reply:
Thanks for these suggestions! I'm still getting adjusted to Rust and wasn't familiar with the `matches!` macro, but it seems really useful